### PR TITLE
Resilience against missing paths

### PR DESCRIPTION
**Pull request by mottosso**
Related to #422.

This change changes the return value of `avalon.pipeline.get_representation_path()`, which returns a path from (1) representation context, (2) global config or (3) representation data, in that order.

- Previously, if the file didn't exist, it would continue checking the next source.
- Currently, a path is returned regardless of whether it exists on disk.

With this change, it's possible to *attempt* to open a missing path, which is then handled by the tool. It's also possible to copy the proposed path to the clipboard via the Loader, to see what the attempted path actually was.

Before, if a path didn't exist, there was no way of telling why, where it looked and what to do about it.

This still isn't perfect however, there's an architectural flaw; the function is almost able to operate entirely in isolation, given only the data present in the representation. That's easily testable and stable, but then there's is one feature that reaches into the database and breaks testability, which is the `io.parenthood` function.

What we could do instead, is separate the global config from paths stored in the representation. That way a tool can ask "Is there a path in this representation? No? Ok, look globally" and handle each error individually.

### CODE REVIEW

**Comment by mkolar** [comment]
So the changed logic now is that if `path_from_representation` was successful (in a sense that it returned something), we won't be checking the next option, but rely on the path returned to be correct.  Right?

I don't see problem with that.  The truth is that if it finds `template` and `context`, fills the path, and that pat doesn't exists it would be good to know about it rather than falling back onto `baked` path right away. That has already caused some minor issues for us in the past. 

Thumbs up from me.

**Comment by mottosso** [comment]
@davidlatwe Please do! Thanks guys.

File: avalon/pipeline.py [hunk 1]
```diff
@@ -1269,16 +1268,14 @@ def path_from_config():
             log.debug("Template references unavailable data: %s" % e)
             return None
 
-        if os.path.exists(path):
-            return os.path.normpath(path)
+        return os.path.normpath(os.path.dirname(path))
```
Comment (mkolar) [inline avalon/pipeline.py]: @iLLiCiTiT noticed that this is actually returning only the folder and not the full path. That will break all the loading won't it?

Comment (davidlatwe) [reply avalon/pipeline.py]: O.o Good spot !
Looks like something happened while merging #371, but wasn't able to figure out how it changed to return dir path.

**Comment by mkolar** [comment]
I'm wondering why this froze in space a bit. I reckon @mottosso simply didn't find time to change it from directory to the full path? Should we just fix it and merge?

**Comment by BigRoy** [comment]
> Should we just fix it and merge?

I suppose so, yes. 

It could still be worth opening an additional issue for the future about what's stated in the initial comment:

> This still isn't perfect however, there's an architectural flaw; the function is almost able to operate entirely in isolation, given only the data present in the representation. That's easily testable and stable, but then there's is one feature that reaches into the database and breaks testability, which is the io.parenthood function.
>
> What we could do instead, is separate the global config from paths stored in the representation. That way a tool can ask "Is there a path in this representation? No? Ok, look globally" and handle each error individually.

Just to be sure we keep that in mind?

**Comment by mkolar** [comment]
We're happy to fix this, I'm just not sure what is the best way. We can either make new PR that does the same thing without the issue of not returning the full path, or send a PR to @mottosso fork which will trickle down here.

**Comment by mottosso** [comment]
New PR is fine, this one is stale.

**Pull request closed without merging**